foreign_cc/extensions.bzl

```diff
@@ -23,22 +23,31 @@ def _init(module_ctx):
         register_built_pkgconfig_toolchain = True,
     )
 
-    versions = {
-        "cmake": _DEFAULT_CMAKE_VERSION,
-        "ninja": _DEFAULT_NINJA_VERSION,
-    }
+    cmake_version = _DEFAULT_CMAKE_VERSION
+    ninja_version = _DEFAULT_NINJA_VERSION
 
+    # Traverse all modules starting from the root one (the first in
+    # module_ctx.modules). The first occurrence of cmake or ninja tag wins.
+    # Multiple versions requested from the same module are rejected.
     for mod in module_ctx.modules:
-        if not mod.is_root:
```

Comment:
We need to maintain this check - only the root module should declare the versions of the tooling.
Good catch though on this being inverted; this should be `if mod.is_root`

Reply:
The way it is written in this PR, we allow any module to declare a version of the tooling but the module that is closest to the root has the precedence. Is there a reason to strictly limit these version declarations to the root module only?